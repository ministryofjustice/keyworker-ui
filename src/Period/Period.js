import React from 'react'
import PropTypes from 'prop-types'
import Input from '@govuk-react/input'
import Select from '@govuk-react/select'
import Header from '@govuk-react/header'

import FilterStyled from './Period.styles'

const Period = ({ duration, period, onInputChange, onButtonClick }) => (
  <form>
    <Header level={3} size="SMALL">
      Select period to view
    </Header>
    <FilterStyled>
      <Input
        name="duration"
        data-qa="keyworker-dashboard-duration"
        value={duration}
        onChange={e => onInputChange({ period, duration: e.target.value })}
      />
      <Select
        name="period"
        data-qa="keyworker-dashboard-period"
        input={{
          value: period,
          onChange: e => onInputChange({ duration, period: e.target.value }),
        }}
      >
        <option value="week">Weekly</option>
        <option value="month">Monthly</option>
        <option value="year">Yearly</option>
      </Select>

      <button type="submit" className="button greyButton" onClick={() => onButtonClick({ duration, period })}>
        Update
      </button>
    </FilterStyled>
  </form>
)
Period.propTypes = {
  onButtonClick: PropTypes.func.isRequired,
  onInputChange: PropTypes.func.isRequired,
  duration: PropTypes.number.isRequired,
  period: PropTypes.string.isRequired,
}

export default Period
