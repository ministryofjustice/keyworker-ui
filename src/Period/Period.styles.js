import styled from 'styled-components'
import { SPACING, FONT_SIZE, BREAKPOINTS } from '@govuk-react/constants'

export const FilterStyled = styled('div')`
  display: flex;
  align-items: flex-end;
  margin-bottom: ${SPACING.SCALE_5} !important;
`
