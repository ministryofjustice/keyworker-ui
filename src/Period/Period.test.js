import React from 'react'
import { shallow } from 'enzyme'
import Period from './Period'

describe('<Period />', () => {
  const props = {
    onButtonClick: jest.fn(),
    onInputChange: jest.fn(),
    fromDate: '2018-12-01',
    toDate: '2018-12-31',
  }

  it('renders without crashing', () => {
    shallow(<Period {...props} />)
  })

  it('should render the default from date in the correct format', () => {
    const wrapper = shallow(<Period {...props} />)
    const fromDate = wrapper.findWhere(node => node.props().inputId === 'keyWorkerStatsFromDate')

    expect(fromDate.prop('defaultValue')).toEqual('01/12/2018')
  })

  it('should render the default to date in the correct format', () => {
    const wrapper = shallow(<Period {...props} />)
    const toDateInput = wrapper.findWhere(node => node.props().inputId === 'keyWorkerStatsToDate')

    expect(toDateInput.prop('defaultValue')).toEqual('31/12/2018')
  })
})
