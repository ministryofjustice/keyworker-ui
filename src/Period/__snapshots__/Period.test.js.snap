// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Period /> should render the default duration and period 1`] = `
<form>
  <h3
    className="e1vcssft0 css-l3100a-StyledHeader-StyledHoc e4u1u2g0"
    size="SMALL"
  >
    Select period to view
  </h3>
  <div
    className="css-1pw7kel"
  >
    <input
      className="e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
      data-qa="keyworker-dashboard-duration"
      name="duration"
      onChange={[Function]}
      type="text"
      value={4}
    />
    <label
      className="e1vcssft0 e1vcssft0 css-15b2i3k-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
      data-qa="keyworker-dashboard-period"
      name="period"
    >
      <span
        className="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
      />
      <select
        className="css-pvtrvx-StyledSelect e4u1u2g0"
        onChange={[Function]}
        value="week"
      >
        <option
          value="week"
        >
          Weekly
        </option>
        <option
          value="month"
        >
          Monthly
        </option>
        <option
          value="year"
        >
          Yearly
        </option>
      </select>
    </label>
    <button
      className="button greyButton"
      onClick={[Function]}
      type="submit"
    >
      Update
    </button>
  </div>
</form>
`;
